test/test_parse_osmchange_input.cpp

```diff
@@ -412,6 +424,30 @@ TEST_CASE("Create way, node ref missing", "[osmchange][way][xml]") {
     R"(<osmChange><create><way changeset="858" id="-1"><nd ref="1"/><nd /><tag k="key" v="value"/></way></create></osmChange>)"), http::bad_request);
 }
 
+TEST_CASE("Delete way, no version", "[osmchange][way][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><delete><way changeset="858" id="-1"/></delete></osmChange>)"),
+    http::bad_request);
+}
+
+TEST_CASE("Delete way, no id", "[osmchange][way][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><delete><way changeset="858" version="1"/></delete></osmChange>)"),
+    http::bad_request);
+}
+
+TEST_CASE("Delete way, no changeset", "[osmchange][way][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><delete><way id="-1 version="1"/></delete></osmChange>)"),
```

Comment:
I think there's a quotation mark missing right after `<way id="-1`  - similar issue in line 447 below